web/src/main/java/org/wso2/testgrid/web/operation/ConfigXmlGenerator.java

```diff
@@ -0,0 +1,85 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.operation;
+
+import org.apache.hc.client5.http.fluent.Request;
+import org.wso2.testgrid.web.bean.TestPlanRequest;
+
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_HOME;
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_USER_AUTH_KEY;
+
+import java.io.IOException;
+
+import static javax.ws.rs.core.HttpHeaders.USER_AGENT;
+
+/**
+ * Class of ConfigXmlGenerator which retrieve an existing config.xml from Jenkins and prepare new config.xml
+ * to new job.
+ */
+public class ConfigXmlGenerator {
+
+    /**
+     * Returns the configuration file for the testPlanRequest which can be used to create new Jenkins job.
+     * @param testPlanRequest request for creating new test plan.
+     * @return configuration file for Jenkins job.
+     */
+    public String getConfigXml(TestPlanRequest testPlanRequest) throws Exception {
+        String template = receiveConfigXmlFromJenkins();
+        String[][] replacements = {
+                {"$productName", testPlanRequest.getProductName()},
+                {"$productVersion", "deprecated"}, //Since version and channel is planned to be removed from TestGrid.
+                {"$productChannel", "deprecated"},
+                {"$infrastructureRepo", "\"" + testPlanRequest.getInfrastructure().getRepository() + "\""},
+                {"$deploymentRepo", "\"" + testPlanRequest.getDeployment().getRepository() + "\""},
+                {"$scenariosRepo", "\"" + testPlanRequest.getScenarios().getRepository() + "\""},
+                {"$infraLocation", "\"" + testPlanRequest.getInfrastructure().getRepository() + "\""},
+                {"$deploymentLocation", "\"" + testPlanRequest.getDeployment().getRepository() + "\""},
+                {"$scenariosLocation", "\"" + testPlanRequest.getScenarios().getRepository() + "\""}
+        };
+        return mergeTemplate(template, replacements);
+    }
+
+    /**
+     * Receives configuration file of the template job in Jenkins server.
+     * @return configuration file of the template job.
+     */
+    private String receiveConfigXmlFromJenkins() throws IOException {
+        try {
+            return Request.Get(JENKINS_HOME + "/job/velocityTemplateFolder/job/velocityTemplateJob/config.xml")
+                    .addHeader("User-Agent", USER_AGENT)
+                    .addHeader("Authorization", "Basic " + JENKINS_USER_AUTH_KEY)
+                    .execute().returnContent().asString().trim();
+        } catch (IOException e) {
+            throw new IOException("Request failed while accessing $url: " + e.toString(), e);
```

Comment:
You are not printing the actual URL here! :)

Reply:
Changed.